Source/WebCore/html/canvas/PlaceholderRenderingContext.cpp

```diff
@@ -61,21 +64,38 @@ void PlaceholderRenderingContextSource::setPlaceholderBuffer(ImageBuffer& imageB
     std::unique_ptr serializedClone = ImageBuffer::sinkIntoSerializedImageBuffer(WTFMove(clone));
     if (!serializedClone)
         return;
-    callOnMainThread([weakPlaceholder = m_placeholder, buffer = WTFMove(serializedClone)] () mutable {
-        RefPtr placeholder = weakPlaceholder.get();
-        if (!placeholder)
+    callOnMainThread([protectedThis = Ref { *this }, this, buffer = WTFMove(serializedClone), delegateCopiedToLayer] () mutable {
+        assertIsMainThread();
```

Comment:
Seems like overkill to do this assertion given it’s a `callOnMainThread`. Why wouldn’t we have this inside every block we ever pass to `callOnMainThread`?